Source/WebCore/platform/graphics/filters/FilterEffect.cpp

```diff
@@ -50,15 +50,26 @@ FilterImageVector FilterEffect::takeImageInputs(FilterImageVector& stack) const
     return inputs;
 }
 
-FloatRect FilterEffect::calculatePrimitiveSubregion(const Filter& filter, const FilterImageVector& inputs, const std::optional<FilterEffectGeometry>& geometry) const
+static Vector<FloatRect> inputPrimitiveSubregions(const FilterImageVector& inputs)
+{
+    Vector<FloatRect> inputPrimitiveSubregions;
+    inputPrimitiveSubregions.reserveInitialCapacity(inputs.size());
+
+    for (auto& input : inputs)
+        inputPrimitiveSubregions.uncheckedAppend(input->primitiveSubregion());
+
+    return inputPrimitiveSubregions;
```

Comment:
You could write this as:

```c++
    return inputs.map([] (auto& input) { return input->primitiveSubregion(); });
```

if you think it's clearer. (I'm always on the fence, given C++'s verbose lambda syntax.)